library/src/main/java/com/anjlab/android/iab/v3/PurchaseInfo.java

```diff
@@ -108,4 +113,15 @@ public PurchaseInfo createFromParcel(Parcel source)
                     return new PurchaseInfo[size];
                 }
             };
+
+    @Override
+    public boolean equals(Object o)
```

Comment:
Check if this is good for equals or if other values should be checked. I replaced the test above to use equals rather than checking individual attributes.